generator/generator_tests/osm_type_test.cpp

```diff
@@ -3061,6 +3050,83 @@ UNIT_CLASS_TEST(TestWithClassificator, OsmType_ComplexTypesSmoke)
   }
 }
 
+UNIT_CLASS_TEST(TestWithClassificator, OsmType_HighwayTypesConversion)
+{
+  using Type = std::vector<std::string>;
+  std::vector<std::pair<Type, Tags>> const conversions = {
+    {{"highway", "cycleway"}, {{"highway", "path"}, {"foot", "no"}, {"bicycle", "yes"}}},
+
+    // Paved etc. paths to footways.
+    {{"highway", "footway"}, {{"highway", "path"}, {"surface", "paved"}}},
+    {{"highway", "footway"}, {{"highway", "path"}, {"surface", "paved"}, {"smoothness", "bad"}}},
+    {{"highway", "footway"}, {{"highway", "path"}, {"surface", "compacted"}, {"smoothness", "intermediate"}}},
+    {{"highway", "footway"}, {{"highway", "path"}, {"surface", "gravel"}, {"smoothness", "good"}}},
```

Comment:
path + compacted/grave/fine_gravel is definitely good (or even designated) for bicycles. I suspect that there are tons of ways with path + compacted in OSM.

Current footway bicycle speed = 2.
Current path bicycle speed = 6 (ok, maybe 0.8 factor for compacted/gravel).

Reply:
compacted+gravel paths are less than 300k cases which is ~2% of all paths (roughly same amount as surface=grass)

Reply:
```BicycleModel::AllLimitsInstance()``` returns speed with ```bicycle_model::NormalPedestrianAndFootwaySpeed()```

In reality, it depends on a country, but in most countries footway speed is _dismount_

Reply:
Today 300K, but tomorrow? I strongly believe that half of all paths are _compacted/fine_gravel_ by default, and tomorrow they all will be tagged like this :)

Reply:
the idea here is that paths like this are perceived by users more like footways (something specially prepared/formed) and hence better shown as white

![IMG_20240419_182004_sm.jpg](https://github.com/organicmaps/organicmaps/assets/18434508/f7e1f7ab-c897-49ec-a3a0-ff7264ab16f4)

Reply:
> Today 300K, but tomorrow? I strongly believe that half of all paths are _compacted/fine_gravel_ by default, and tomorrow they all will be tagged like this :)

Impossible. The most of the paths are mapped in outdoors and doesn't have a special surface whereas compacted or a smooth gravel layer is a specially prepared surface, one can find it in urban parks mostly.

Reply:
> `BicycleModel::AllLimitsInstance()` returns speed with `bicycle_model::NormalPedestrianAndFootwaySpeed()`
> 
> In reality, it depends on a country, but in most countries footway speed is _dismount_

I assume that in countries where footways by default imply bicycle=no mappers do not rely on `path` having implicit bicycle=yes, but are more accustomed to add a specific `bicycle=` tag explicitly...

And I totally agree that bicycle routing should be improved with better bicycle-specific grading, but the way I see it - it requires a big refactoring... and I don't see an easy way to "adjust the routing" for this particular case.

E.g. ATM the router prefers `path unpaved_bad yesbicycle  		[ weight:8.1, eta:5.4 ]` to good paved paths without bicycle=yes! So compared to this you're really nit-picking here IMO :)

I don't mind to remove conversion of compacted/gravel, I just added it specifically because IMO it'd make things look more consistent..

Reply:
If later we have reports of this compacted/gravel conversion is making trouble for bicycle routing then we can reconsider it.